backend/app/api/routes/podcasts.py

```diff
@@ -0,0 +1,151 @@
+import os
+import uuid
+from typing import Any, Optional, List
+
+from fastapi import APIRouter, HTTPException
+from fastapi.responses import StreamingResponse, JSONResponse
+from pydantic import BaseModel
+
+from app.api.deps import CurrentUser, SessionDep
+from sqlmodel import select
+from sqlalchemy.orm import selectinload
+from app.core.config import settings
+from app.models.podcast import Podcast
+from app.schemas.public import PodcastPublic, PodcastsPublic
+from app.services.podcast_service import generate_podcast_for_course
+
+router = APIRouter(prefix="/podcasts", tags=["podcasts"])
+
+
+@router.get("/{course_id}", response_model=PodcastsPublic)
+def list_podcasts(course_id: uuid.UUID, session: SessionDep, current_user: CurrentUser) -> Any:
+    pods = session.exec(select(Podcast).where(Podcast.course_id == course_id)).all()
+    return PodcastsPublic(data=[PodcastPublic.model_validate(p) for p in pods])
+
+
+class GeneratePodcastRequest(BaseModel):
+    title: Optional[str] = None
+    mode: Optional[str] = None  # 'dialogue' | 'presentation'
+    topics: Optional[str] = None
+    teacher_voice: Optional[str] = None
+    student_voice: Optional[str] = None
+    narrator_voice: Optional[str] = None
+    document_ids: Optional[List[uuid.UUID]] = None
+
+
+@router.post("/{course_id}/generate", response_model=PodcastPublic)
+async def generate_podcast(
+    course_id: uuid.UUID,
+    session: SessionDep,
+    current_user: CurrentUser,
+    body: GeneratePodcastRequest | None = None,
```

Comment:
If the body is required for this route, why default it to None? When required, all clients consuming this route must include it, as a validation is generated when creating the client folder based on the backend schema.